lib/libzfs/libzfs_import.c

```diff
@@ -248,6 +248,17 @@ is_mpath_whole_disk(const char *path)
 	return (wholedisk);
 }
 
+static int
+udev_device_is_ready(struct udev_device *dev)
+{
+#ifdef HAVE_LIBUDEV_UDEV_DEVICE_GET_IS_INITIALIZED
+	return (udev_device_get_is_initialized(dev));
+#else
+	/* wait for DEVLINKS property to be initialized */
+	return (udev_device_get_property_value(dev, "DEVLINKS") != NULL);
```

Comment:
Yup, sounds good we can definitely revisit this.  Although we're always going to need to keep the polling behavior as a fallback for compatibility reasons since we can't assume every system has udev monitoring and we need to provide at least basic functionality.